pkg/urbit/jets/e/scow.c

```diff
@@ -188,202 +753,260 @@ _print_p(u3_atom cor, u3_atom p)
 
-  return list;
+  u3z(ud);
+  return aus;
 }
 
+#undef DOT
+
+#define DOT 0x2e
+#define ZED 0x30
+#define VEE 0x76
+
 static
 u3_noun
 _print_uv(u3_atom uv)
 {
   // number of characters printed "between" periods.
   c3_i between = 0;
-  u3_noun list = 0;
+  u3_noun aus = 0;
+  c3_w wrt = 0;
 
   // increase input refcount to be consumed in u3ka_div(), which will free each
   // intermediary state.
   u3k(uv);
 
   do {
     if (between == 5) {
-      list = u3nc('.', list);
+      aus = u3qa_add(u3qc_lsh(2, 2, aus), DOT);
       between = 0;
     }
 
     c3_y tig = u3qa_mod(uv, 32);
-    list = u3nc(to_digit(tig), list);
+    wrt = to_digit(tig);
+    aus = u3qa_add(u3qc_lsh(2, 2, aus), wrt);
+    //u3m_p("0v> wrt:", wrt);
+    //u3m_p("0v> aus:", aus);
     between++;
     uv = u3ka_div(uv, 32);
   } while (uv != 0);
 
-  return u3nt('0', 'v', list);
+  aus = u3qa_add(u3qc_lsh(2, 2, aus), VEE);
+  aus = u3qa_add(u3qc_lsh(2, 2, aus), ZED);
+  //u3m_p("0v> aus:", aus);
+  u3z(uv);
+  return aus;
 }
 
+#undef DOT
+#undef ZED
+#undef VEE
+
+#define DOT 0x2e
+#define ZED 0x30
+#define WEE 0x77
+
 static
 u3_noun
 _print_uw(u3_atom uw)
 {
   // number of characters printed "between" periods.
   c3_i between = 0;
-  u3_noun list = 0;
+  u3_noun aus = 0;
+  c3_w wrt = 0;
 
   // increase input refcount to be consumed in u3ka_div(), which will free each
   // intermediary state.
   u3k(uw);
 
   do {
     if (between == 5) {
-      list = u3nc('.', list);
+      aus = u3qa_add(u3qc_lsh(2, 2, aus), DOT);
       between = 0;
     }
 
     c3_y tig = u3qa_mod(uw, 64);
-    list = u3nc(to_w_digit(tig), list);
+    wrt = to_w_digit(tig);
+    aus = u3qa_add(u3qc_lsh(2, 2, aus), wrt);
+    u3m_p("0w> wrt:", wrt);
     between++;
     uw = u3ka_div(uw, 64);
   } while (uw != 0);
 
-  return u3nt('0', 'w', list);
+  aus = u3qa_add(u3qc_lsh(2, 2, aus), WEE);
+  aus = u3qa_add(u3qc_lsh(2, 2, aus), ZED);
+  u3z(uw);
+  return aus;
 }
 
+#undef DOT
+#undef ZED
+#undef WEE
+
 u3_noun
 u3we_scow(u3_noun cor)
 {
   u3_atom mod;
   u3_atom atom;
 
   if (c3n == u3r_mean(cor, u3x_sam_2, &mod,
-                      u3x_sam_3, &atom, 0)) {
+                           u3x_sam_3, &atom, 0)) {
     return u3m_bail(c3__exit);
   }
 
+  u3m_p(">mod", mod);
+  u3m_p(">atom", atom);
+
+  u3_noun cord;
   switch (mod) {
     case c3__da:
-      return _print_da(cor, atom);
+      cord = _print_da(cor, atom);
+      break;
 
     case 'p':
-      return _print_p(cor, atom);
+      cord = _print_p(cor, atom);
+      break;
 
     case c3__ud:
-      return _print_ud(atom);
+      cord = _print_ud(atom);
+      break;
 
     case c3__uv:
-      return _print_uv(atom);
+      cord = _print_uv(atom);
+      break;
 
     case c3__uw:
-      return _print_uw(atom);
+      cord = _print_uw(atom);
+      break;
 
     default:
       return u3_none;
   }
+
+  if (cord == u3_none) {
+    return cord;
+  }
+  u3_noun ret = u3qe_trip(cord);
+  //u3z(mod);
+  //u3z(atom);
+  return ret;
```

Comment:
This leaks `cord` -- `u3qe_trip()` has `RETAIN` refcount semantics (we this because it's prefixed with `u3q`), but `cord` is an allocation from this function, so this function is still responsible for it.